components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Modal} from 'react-bootstrap';
+
+import {AdminConfig} from 'mattermost-redux/types/config';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {Constants} from 'utils/constants';
+import {localizeMessage} from 'utils/utils.jsx';
+import {t} from 'utils/i18n';
+
+const INT32_MAX = 2147483647;
+
+type Props ={
+    config: Partial<AdminConfig>;
+    show: boolean;
+    onClose: () => void;
+    actions: {
+        updateConfig: (config: AdminConfig) => ActionFunc & Partial<{error?: ClientErrorPlaceholder}>;
+        getConfig: () => ActionFunc;
+    };
+}
+
+type ClientErrorPlaceholder = {
+    message: string;
+    server_error_id: string;
+}
+
+export default function EditPostTimeLimitModal(props: Props) {
+    const [saving, setSaving] = React.useState(false);
+    const [errorMessage, setErrorMessage] = React.useState('');
+    const [postEditTimeLimit, setPostEditTimeLimit] = React.useState<number | string>(parseInt(props.config.ServiceSettings!.PostEditTimeLimit.toString(), 10));
+
+    React.useEffect(() => {
+        props.actions.getConfig();
+    }, []);
+
+    const save = async () => {
+        setSaving(true);
+        setErrorMessage('');
+
+        const val = parseInt(postEditTimeLimit.toString(), 10);
+        if (val !== Constants.UNSET_POST_EDIT_TIME_LIMIT) {
+            if (val.toString() === 'NaN' || val <= 0 || val > INT32_MAX) {
+                setErrorMessage(localizeMessage('edit_post.time_limit_modal.invalid_time_limit', 'Invalid time limit'));
+                setSaving(false);
+                return false;
+            }
+        }
+
+        const newConfig = JSON.parse(JSON.stringify(props.config));
+        newConfig.ServiceSettings.PostEditTimeLimit = val;
+
+        const {error} = await props.actions.updateConfig(newConfig);
+        if (error) {
+            setErrorMessage(error.message);
+            setSaving(false);
+        } else {
+            setSaving(false);
+            props.onClose();
+        }
+
+        return true;
+    };
+
+    const handleOptionChange = (e: React.FormEvent<HTMLInputElement>) => {
+        const {value} = e.currentTarget;
+        if (value === Constants.ALLOW_EDIT_POST_ALWAYS) {
+            setPostEditTimeLimit(Constants.UNSET_POST_EDIT_TIME_LIMIT);
+        } else {
+            setPostEditTimeLimit('');
+        }
+    };
```

Comment:
instead of flagging the `disable` by using a negative number (constant) we should rather add a new state (e.g. `setAlwaysAllowPostEditing`) ... this has the benefit that the value stored for the input field can be persisted and does not need to be reset to a default value every time.

Another benefit is we can massively simplify that handler:

```suggestion
    const handleOptionChange = ({currentTarget}: React.FormEvent<HTMLInputElement>) => setAlwaysAllowPostEditing(currentTarget.value === Constants.ALLOW_EDIT_POST_ALWAYS);
```

Reply:
I agree with `setAlwaysAllowPostEditing`. However we would still need to set `setPostEditTimeLimit` to -1. 

Because in the [EditPostTimeLimitButton](https://github.com/mattermost/mattermost-webapp/blob/8d47e8e1e035f042288fc17862241e6682cc76a8/components/admin_console/permission_schemes_settings/edit_post_time_limit_button/edit_post_time_limit_button.jsx#L20) component, We are checking if `PostEditTimeLimit` in `GlobalState` is set to `-1`. 

Of course we can also change that logic as well but I believe, we already achieved good amount of simplification here. Please let me know what do you think?